code/modules/paperwork/paper.dm

```diff
@@ -299,9 +298,95 @@
+				"<span class='warning'>You try to burn the [src.name], but burn your hand trying!</span>")
+			return //you fail before even managing to burn the paper!
+	if(prot) //user is human and is protected from fire, let's make them a badass
+		user.visible_message( \
+			"<span class='warning'>[user] holds up the [src.name] and sets it on fire, holding it in \his hand as it burns down to ashes. Damn, \he's cold.</span>", \
+			"<span class='warning'>You hold up the [src.name] and set it on fire, holding it in your hand as it burns down to ashes. Damn, you're cold.</span>")
+	else
+		user.visible_message( \
+			"<span class='warning'>[user] holds up the [src.name] and sets it on fire, reducing it to a heap of ashes.</span>", \
+			"<span class='warning'>You hold up the [src.name] and set it on fire, reducing it to a heap of ashes.</span>")
+	new ashtype(get_turf(src)) //not using ashify() since it calls for src.loc rather than get_turf(src), and requires the object to be on fire also
+	qdel(src)
 	return
 
+var/global/list/paper_folding_results = list ( \
+	"paper plane" = /obj/item/weapon/p_folded/plane,
+	"paper hat" = /obj/item/weapon/p_folded/hat,
+	"ball of paper" = /obj/item/weapon/p_folded/ball,
+	"folded note" = /obj/item/weapon/p_folded/note_small,
+	"origami crane" = /obj/item/weapon/p_folded/crane,
+	"origami boat" = /obj/item/weapon/p_folded/boat,
+	"origami heart" = /obj/item/weapon/p_folded/folded_heart,
+	)
+
+/obj/item/weapon/paper/verb/fold()
+	set category = "Object"
+	set name = "Fold paper"
+	set src in usr
+
+	if (!canfold(usr)) return
+	. = paper_folding_results[(input("What do you want to make the paper into?", "Paper Folding") as null|anything in paper_folding_results)]
+	if (. == null) return
+	if (!canfold(usr)) return //second check in case some chucklefuck moves the paper or falls down while the menu is open
+
+	usr.drop_item(src)	//Drop the original paper to free our hand and call proper inventory handling code
+	var/obj/item/weapon/p_folded/P = new .(get_turf(usr)) 	//Let's make a new item
+	P.unfolded = src										//that unfolds into the original paper
+	src.loc = P												//and also contains it, for good measure.
+	usr.put_in_hands(P)
+	P.pixel_y = src.pixel_y
+	P.pixel_x = src.pixel_x
+	if (istype(src, /obj/item/weapon/paper/nano))
+		P.color = "#9A9A9A"
+		P.nano = 1
+	usr.visible_message("<span class='notice'>[usr] folds \the [src.name] into a [P.name].</span>", "<span class='notice'>You fold \the [src.name] into a [P.name].</span>")
+	P.add_fingerprint(usr)
+	return
+
+/obj/item/weapon/paper/proc/canfold(mob/user)
+	if(!user)
+		return 0
+	if(user.stat || user.restrained())
+		usr << "<span class='notice'>You can't do that while restrained.</span>"
+		return 0
+	if(user.l_hand != src && user.r_hand != src)
+		usr << "<span class='notice'>You'll need [src] in your hands to do that.</span>"
+		return 0
+	return 1
+
+/obj/item/weapon/paper/verb/wipe_color() //since we don't have erasers or anything
```

Comment:
Rather than doing this (and going through the rigmarole of adding and removing verbs, euch) just add an eraser.

Nothing like this should really be done through verbs, it makes no sense.